manifests/params.pp

```diff
@@ -63,7 +63,7 @@
   $foreman_service = 'foreman'
   $foreman_service_ensure = 'running'
   $foreman_service_enable = true
-  $foreman_service_puma_threads_min = 0
+  $foreman_service_puma_threads_min = 16
```

Comment:
How about we make it dynamic and set it to undef:
```suggestion
  $foreman_service_puma_threads_min = undef
```

Then use `$min = pick($foreman::foreman_service_puma_threads_min, $foreman::foreman_service_puma_threads_max)`. That way users don't need to specify it twice if they want them to match.

You probably also need to change the type to `Optional[Integer[0]]`.

Also an installer migration that clears the answer if they match will be nice for users.

Reply:
I like the dynamic aspect from a code perspective. I am wondering if this will confuse users of the installer who look at it and see that the minimum is set to nothing. Asking out loud if that will lead to support posts or users to think that the minimum is therefore 0 if blank.

Reply:
We can document this in the parameter help that the default (undef) means matching max threads.

Reply:
Ehh, I'll update the documentation of the parameter and let that suffice.